### PR TITLE
Dotcomponents Data Model V2 - structural changes

### PR DESCRIPTION
**Pull request by MatthewJWalls**
## What does this change?

Alters the structure of the dotcomponents data model to be more logically organised

See comments for more discussion

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Comment by MatthewJWalls** [comment]
# The proposed structure

[full example json](https://gist.github.com/MatthewJWalls/bd344fac8d84e60734c305eb1d389192)

## Top Level

At the top level the structure is divided into three broad scopes

```
{
    "page": { ... }
    "site": { ... }
    "config": { ... }
}
```

* page is for data that scopes to the page content itself, such as the article content
* site is for data that scopes to the guardian site, such as the nav
* config is for data that scopes to the application, such as api endpoints

## Page Level

field |  type/example
-- | --
conent | Object
tags | Object
author | "Benjamin Haas"
pageId | "world/2018/sep/13/give-pizza-a-chance-south-koreans-pa-weight-to-thwart-conscription"
pillar | "News"
webPublicationDate | 1536812845000
webPublicationDateDisplay | "Thu 13 Sep 2018 05.27 BST"
section | "world"
webTitle | "Give pizza a chance: South Koreans pack on weight to thwart conscription"
contentId | "world/2018/sep/13/give-pizza-a-chance-south-koreans-pa-weight-to-thwart-conscription"
editionId | "UK"
edition | "UK edition"
contentType | "Article"
subMetaLinks | Object
webURL | "https://www.theguardian.com/world/2018/sep/13/give-pizza-a-chance-south-koreans-pa-weight-to-thwart-conscription"
meta | Object

## Site Level

field |  type/example
-- | --
nav | Object
readerRevenueLinks | Object

## Config Level

field |  type/example
-- | --
ajaxUrl | xxxxx
guardianBaseURL | xxxxx
sentryHost | xxxxx
sentryPublicApiKey | xxxxx
switches | Object
beaconUrl | xxxxx

**Comment by MatthewJWalls** [comment]
This PR should not be merged until we've reviewed the corresponding PR on dotcom-rendering

**Comment by MatthewJWalls** [comment]
rebased to bring in line with Nic's trailtext/structureddata pr

**Comment by SiAdcock** [comment]
My initial thought is that the `site` object looks a bit sparse. Once we move `nav` to Configuration, it's likely that `readerRevenueLinks` could follow, or could otherwise find a more suitable home. So I guess I'm just flagging the possibility that `site` is a temporary thing, until we find a better way of passing this data around. Does that sound fair?

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala [hunk 1]
```diff
@@ -61,68 +66,82 @@ case class ReaderRevenueLinks(
...
+
+case class Content(
+  headline: String,
+  standfirst: Option[String],
+  main: String,
+  body: String,
+  blocks: Blocks,
+  byline: String,
+  trailText: String
 )
```
Comment (SiAdcock) [inline article/app/model/dotcomponents/DotcomponentsDataModel.scala]: Shall we move this up, to keep _The Big Three_ together?

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala [hunk 2]
```diff
@@ -154,17 +181,16 @@ object ReaderRevenueLinks {
...
+  implicit val writes = Json.writes[DCSite]
 }
 
-object Config {
-  implicit val writes = Json.writes[Config]
+object DCConfig {
+  implicit val writes = Json.writes[DCConfig]
 }
 
 object DotcomponentsDataModel {
```
Comment (SiAdcock) [inline article/app/model/dotcomponents/DotcomponentsDataModel.scala]: Should we update this to `VERSION = 2`?

Comment (MatthewJWalls) [reply article/app/model/dotcomponents/DotcomponentsDataModel.scala]: Yep, nice spot thanks will update

**Comment by MatthewJWalls** [comment]
I've combined site and config into one object now

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala (hunk 1 above)
Comment (MatthewJWalls) [reply article/app/model/dotcomponents/DotcomponentsDataModel.scala]: Big two now, but yes I've moved this so page/site classes are together

File: article/conf/schema/dotcomponentsDataModelV2.jsonschema [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "properties": {
+        "page": {
+
+        },
+        "site": {
+
+        },
+        "config": {
+
+        },
```
Comment (SiAdcock) [inline article/conf/schema/dotcomponentsDataModelV2.jsonschema]: We can ditch the `config` property now?

**Review by SiAdcock — APPROVED** [review]
One more small thing, otherwise this is a huge and much-needed structural improvement!

**Pull request merged**